### DISCUSSION TITLE
Interest in iOS

### DISCUSSION THREAD

**Discussion by astubenbord** [astubenbord/paperless-mobile#35] [Polls]
I am interested in the amount of potential users on iOS! If there is enough interest I might consider also publishing to the Apple App Store. Since the costs to publish to app store are quite high (100$/year), this step would only make sense if there are enough users interested and potentially also willing to help me cover part of the fees. 

Thanks for your participation!

**Comment by JigSawFr** [comment]
Or maybe sell the app as a one time purchase ? Not a big fan of subscription but one time purchase will be fine. Otherwise sponsorship on GitHub 😁

**Reply by astubenbord** [reply]
You're right. I will probably try the sponsorship option if this project gains a bit more traction and there are enough people interested! :)

**Reply by astubenbord** [reply]
Quick Update: I've set up GitHub Sponsors with the monthly goal of 8€, which will pay for the annual fee :)

**Comment by lagmoellertim** [comment]
Are there any updates on the iOS version of this app? If buying a developer certificate is not an option for you, I recently discovered a method to distribute iOS apps without publishing it in the app store. Using `altstore`, users can install `.ipa`.Files manually without the store on their iPhone. These files normally also need a certificate, but when installing it using `altstore`, this is not checked. So you can build a simple `.ipa` file by following this [guide](https://stackoverflow.com/a/72724017) and publish it to github. This process could be automated as well using a CI pipeline.

**Reply by astubenbord** [reply]
Interesting. I'm currently setting up a pipeline for automatic distribution to Google play and eventually also GitHub, thanks for the info, I'll have a look.

**Comment by JigSawFr** [comment]
@astubenbord Hey mate, can you add me on Discord to discuss on it further: <redacted>
Thanks

**Reply by astubenbord** [reply]
Sure!

**Comment by JigSawFr** [comment]
Try the app with TestFlight: 
https://testflight.apple.com/join/nmLJvzP9 
Publishing of builds is in manual stage, working to automate it with Github.
Don't hesitate to share your comments guys :)

**Reply by JigSawFr** [reply]
@astubenbord dont hesitate to come on Discord to discuss further 👋

**Reply by gosoares** [reply]
"This beta ins't accepting new testers right now."

**Reply by 2600box** [reply]
Would love to get access to the testflight, thanks!

**Reply by Chluz** [reply]
Same for me, was using this on Android previously and it was really good !

**Reply by JigSawFr** [reply]
Just need to add some things for Apple Review, hope to solve it soon

**Reply by Dom987911** [reply]
Any news?

**Reply by JigSawFr** [reply]
Hope pretty soon, it's ping pong time with Apple Validation Team (yeah.. for TestFlight builds too)

**Reply by JigSawFr** [reply]
@gosoares @Dom987911 @Chluz @2600box Apple validated app for TestFlight, link should work again :)

**Reply by Dom987911** [reply]
At first glance, everything looks good. Only the settings can not be used. When will it officially released in the AppStore?

**Reply by Chluz** [reply]
Same for me, tested basic functions, and no no issues yet (even the document scanning with camera works)

**Reply by JigSawFr** [reply]
Thanks all ! I need to work on automated releasing for TestFlight, but much more complicated than Android. For the moment it's manually published. And after that I will publish release on AppStore and pre-release on TestFlight 😄

**Reply by JigSawFr** [reply]
> At first glance, everything looks good. Only the settings can not be used. When will it officially released in the AppStore? 

Strange it's working on my side, can you elaborate ?

**Reply by Dom987911** [reply]
It only shows an grey screen if I click on Setting.

**Reply by JigSawFr** [reply]
> It only shows an grey screen if I click on Setting.

Can you tell us:
- Version of your paperless instance
- Version of your app (menu -> about this app)
Thanks

**Reply by JigSawFr** [reply]
I've submitted v. 2.2.4 to TestFlight.

**Reply by Chluz** [reply]
> > It only shows an grey screen if I click on Setting.
> 
> Can you tell us:
> 
> * Version of your paperless instance
> * Version of your app (menu -> about this app)

Same grey screen in settings here, and it seems you can't exit from it - need to force close the app
App version 2.1.0 +22 (2.2.4 is not yet available for me)
Paperless ngx version 1.14.5 with client cert authentication.

**Reply by astubenbord** [reply]
The app version 2.1.0 does not yet support paperless 1.14+. However, 2.2.4 does and we're working on the next stable release supporting the features from 1.14.

**Reply by Dom987911** [reply]
> > > It only shows an grey screen if I click on Setting.
> 
> > 
> 
> > Can you tell us:
> 
> > 
> 
> > * Version of your paperless instance
> 
> > * Version of your app (menu -> about this app)
> 
> 
> 
> Same grey screen in settings here, and it seems you can't exit from it - need to force close the app
> 
> App version 2.1.0 +22 (2.2.4 is not yet available for me)
> 
> Paperless ngx version 1.14.5 with client cert authentication.
> 
> 
> 
> 

Same here but you can swipe back.

**Reply by Chluz** [reply]
2.2.4 just became available, and the settings page is now working fine. Thank you devs !

**Reply by boexler** [reply]
Would it be possible to still get access for test flight? I am also extremely interested in the app for IOs 😃💪

**Reply by astubenbord** [reply]
In theory yes, but it seems like @JigSawFr, who currently manages the distribution on iOS slide, is very busy at the moment. I have recently joined the developer program myself, so I'm waiting for a response whether he can transfer the app to my apple developer account so I can manage the updates in the future myself :) I don't really want to publish it separately, but I also don't want to wait that much longer to publish the newer versions, so we'll see.

**Reply by JigSawFr** [reply]
@astubenbord I've read the transfer criteras, and it seems not possible to transfer the app in the current state. I suggest to follow my doc to proceed re-initiation of the app in your current account.

**Reply by JigSawFr** [reply]
I just released the app name to be re-used again @astubenbord ;)

**Reply by astubenbord** [reply]
Thanks! I will do :)

**Reply by astubenbord** [reply]
Seems like the app has to be deleted in order for the app name to become available again? Or do you know how long it may take for it to be available again? Thanks!

**Reply by JigSawFr** [reply]
> Seems like the app has to be deleted in order for the app name to become available again? Or do you know how long it may take for it to be available again? Thanks!

Already done on my side, don't know if there is a timing. UI said if I delete the app it will be available for other's devs :)

**Reply by astubenbord** [reply]
@JigSawFr Hm, seems like the name is still not available. I'll contact the support then I guess, maybe they can resolve this :)

**Comment by Chluz** [comment]
I seem to have been bumped from the testflight beta. Was that intentional ? I was very much enjoying adding files to paperless from the iphone camera.

**Reply by astubenbord** [reply]
No, not intentional. I don't manage the iOS distribution myself, but I'm planning on doing so.

**Comment by akoen** [comment]
Are there any current plans to release paperless mobile on Testflight or on the App Store? I completely understand that the developer account is expensive, but would love to be able to use the app if it's possible.

**Reply by boexler** [reply]
Maybe it's possible to get a free account? https://developer.apple.com/support/membership-fee-waiver/

**Reply by astubenbord** [reply]
The account itself is not a problem, I already paid for it. I'm rather limited by time at the moment and I currently also don't have access to a macbook :/

**Reply by nstrelow** [reply]
Ah dang, I hate Apple for this exclusivity bullshit (have been since 2013, where people kept pestering me about an iOS app).

It might be possible to compile and build version with github actions + fastlane on a free github MacOS machine, at least it was some time ago.

I did set it up here once before: https://github.com/hpi-studyu/studyu/actions/runs/7802327078/job/21279514013
I think this did work without access to a Macbook, but maybe I am missing some initial setup stuff.

**Comment by akoen** [comment]
> The account itself is not a problem, I already paid for it. I'm rather limited by time at the moment and I currently also don't have access to a macbook :/

Totally understand. Let me know if there’s any way I can help with my macbook although GitHub actions does look like a good strategy. You’re also totally justified in charging a small fee for the app on ios to recoup your costs and time.
